topi/python/topi/cuda/conv2d_int8.py

```diff
@@ -0,0 +1,323 @@
+    s[conv].tensorize(rc_block, _dp4a)
+
+    s[AA].compute_at(s[conv], rxo)
+    s[WW].compute_at(s[conv], rxo)
+
+    # cooperative fetching
+    for load in [AA, WW]:
+        if load == AA:
+            n, f, y, x, c = s[load].op.axis
+            if pad_data == packed_data and stride_h == 1 and stride_w == 1:
+                s[load].vectorize(c)
+                fused = s[load].fuse(n, f, y, x)
+            else:
+                c, _ = s[load].split(c, factor=4)
+                fused = s[load].fuse(n, f, y, x, c)
+        else:
+            n, f, y, x, oc_chunk, c = s[load].op.axis
+            fused = s[load].fuse(n, f, y, x, oc_chunk)
+            s[load].vectorize(c)
+
+        fused, tx = s[load].split(fused, factor=cfg["tile_x"].size[2])
+        fused, ty = s[load].split(fused, factor=cfg["tile_y"].size[2])
+        fused, tz = s[load].split(fused, factor=cfg["tile_f"].size[2])
+        s[load].bind(tz, tvm.thread_axis("threadIdx.z"))
+        s[load].bind(ty, tvm.thread_axis("threadIdx.y"))
+        s[load].bind(tx, tvm.thread_axis("threadIdx.x"))
+
+    # double buffer
+    cfg.define_knob('AA_double_buffer', [0, 1])
+    cfg.define_knob('WW_double_buffer', [0, 1])
+    if cfg['AA_double_buffer'].val:
+        s[AA].double_buffer()
+    if cfg['WW_double_buffer'].val:
+        s[WW].double_buffer()
+
+    # unroll
+    cfg.define_knob("auto_unroll_max_step", [0, 512, 1500])
+    s[output].pragma(kernel_scope, 'auto_unroll_max_step',
+                     cfg['auto_unroll_max_step'].val)
+    s[output].pragma(kernel_scope, 'unroll_explicit', False)
+
+    return s
+
+
+@conv2d_NCHWc_int8_prepacked.register(["cuda"])
+@autotvm.task.dispatcher
+def conv2d_NCHWc_int8_prepacked_dispatcher(data, kernel, stride, padding, layout, out_dtype):
+    assert layout == 'NCHW4c'
+    return _conv2d_NCHWc_int8_arg_to_workload(data, kernel, stride, padding, out_dtype)
+
+
+@conv2d_NCHWc_int8_prepacked_dispatcher.register("int8")
+def _decl_conv2d_NCHWc_int8_prepacked(cfg, data, kernel, stride, padding, layout, out_dtype):
+    return conv2d_NCHWc_int8(cfg, data, kernel, stride, padding, layout, out_dtype,
+                             pre_computed=True)
+
+@autotvm.register_topi_schedule(schedule_conv2d_NCHWc_int8_prepacked, ["cuda"], ["int8"])
+def schedule_conv2d_NCHWc_int8_prepacked_cuda(cfg, outs):
+    """TOPI schedule callback of conv2d for cuda
+
+    Parameters
+    ----------
+    cfg: ConfigEntity
+        The config for this template
+
+    outs: Array of Tensor
+        The computation graph description of conv2d
+        in the format of an array of tensors.
+
+    Returns
+    -------
+    s: Schedule
+        The computation schedule for conv2d.
+    """
+    s = tvm.create_schedule([x.op for x in outs])
+
+    def _callback(op):
+        if 'conv2d_NCHWc_int8' in op.tag:
+            schedule_conv2d_NCHWc_int8(cfg, s, op.output(0), pre_computed=True)
+
+    traverse_inline(s, outs[0].op, _callback)
```

Comment:
With bias enabled, I encountered some direct host side access error because bias is not computed inline. @merrymercy Could you help?

Reply:
you need to handle bias and other elemwise op (relu, batch norm, etc) inside conv2d schedule. In cuda schedule, they are handled [here](https://github.com/dmlc/tvm/blob/master/topi/python/topi/cuda/conv2d_direct.py#L42-L45).